frame/lottery/src/benchmarking.rs

```diff
@@ -69,46 +73,68 @@ benchmarks! {
 		Participants::<T>::insert(&caller, already_called);
 
 		let call = frame_system::Call::<T>::remark { remark: vec![] };
-	}: _(RawOrigin::Signed(caller), Box::new(call.into()))
-	verify {
+	
+		#[extrinsic_call]
+		_(RawOrigin::Signed(caller), Box::new(call.into()));
+	
 		assert_eq!(TicketsCount::<T>::get(), 1);
+
+		Ok(())
 	}
 
-	set_calls {
-		let n in 0 .. T::MaxCalls::get() as u32;
+	#[benchmark]
+	fn set_calls(n: Linear<0, 1>) -> Result<(), BenchmarkError> {
```

Comment:
@sam0x17 Need a bit of your help here. The second param of `Linear` should take `T::MaxCalls::get() as u32` but the current macro doesn't seem to support it.

Reply:
@ggwpez how do you feel about this? Do we want to enable scenarios like this where the upper bound is based on something like `T::MaxCalls::get()`? Do we want support for this or is this an anti-pattern?

Reply:
I am ok with `Into<u32>` as the bound. I don't know if we'll still be able to use the const-style syntax we've been using like `Linear<5, 10>` etc.

Now that I'm looking at this again, I think we did the `<lit, lit>` style syntax because we thought we were going to be able to lean on the type system more, but it turns out these types are never instantiated anyway and they are syntactic sugar only, so I don't see why we couldn't do something sexier like `5..10` where we use the `static_assertions` crate to assert that the expressions on either side of the `..` are `Into<u32>`. That would definitely be a better looking syntax anyway

Reply:
OK but that said I remember now it's because in the future we might have non-linear

Reply:
so I don't like a..b because we need to say Linear somewhere and I don't like Linear<a, b> because const params dont support Into<u32> in real life
they can support it here because this is just syntactic sugar, but it feels like we're abusing the syntax, I'd rather make it more obvious if anything that this is a custom syntax... 🤔

Reply:
ok so my thought is actually that 99.99% of the time it is going to be linear anyway, so let's put the burden of verbosity on the "non-linear" syntax if we ever even end up doing that, so let's just interpret `a..b` where `a: Into<u32>` and `b: Into<u32>` as linear over `a..b`. We should also adopt the conventions of `std::ops::Range` and make `a` inclusive and `b` exclusive, since this is how ranges work in Rust.

Then if we ever actually need to add something that _isn't_ linear, we can just figure out a new syntax for those like `NonLinear<a..b>` or something. Let's cross that bridge when we get to it and make the golden path have better dev ux imo

thoughts @ggwpez ?

Reply:
TLDR: my preference would be we just switch to `a..b` where a is `Into<u32>` and be is `Into<u32>` and remove the LInear struct and the `ParamRange` trait entirely and just interpret `a..b` as always linear. Then if we ever add non-linear stuff we can worry about how that should look then

Reply:
I think you can just expand the parsing from `LitInt` to `syn::GenericArgument`.  
Only difficult part would be to print a warning if `start > end`, but could be dont with static-assertions.  

We can also do the `a..b` that you mentioned. The only advantage of the `Linear` is that it is more self documenting and we *could* add more generics in the future, like `step`.